### ISSUE TITLE
Editable non-object JSON roots

### ISSUE THREAD

**Issue by marcklingen** [YYsuni/react18-json-view#36]
Currently `editable` works if the root node is of type object. JSON roots can also be arrays or scalars, these do not seem to be editable in react18-json-view as of now (if I did not do a mistake).

Do you think adding this functionality is reasonable? Otherwise, we would build alternative UI in our project to toggle the JSON editor or default to a regular input field to edit scalars (strings).

Example (https://github.com/langfuse/langfuse)
![CleanShot 2024-02-26 at 16 28 57](https://github.com/YYsuni/react18-json-view/assets/2834609/a3add5ab-20cf-4f2e-9662-8abcdd51d161)

**Comment by YYsuni** [comment]
OK, I will, my heavy account 😂.

**Comment by YYsuni** [comment]
react18-json-view@0.2.8-canary.2 available, Sir. @marcklingen

**Comment by marcklingen** [comment]
Thank you! Just tried it, does not work for me. I upgraded from react18-json-view@0.2.8-canary.1 to react18-json-view@0.2.8-canary.2.

![CleanShot 2024-02-29 at 12 04 53@2x](https://github.com/YYsuni/react18-json-view/assets/2834609/b67f07b3-1c90-4b44-a7fd-9b9be9671c4b)

Implementation
https://github.com/langfuse/langfuse/blob/2f55acec7c06116ea92a19764416c9ff388e3112/web/src/features/datasets/components/EditDatasetItem.tsx#L118-L130

**Comment by YYsuni** [comment]
Please try @0.2.8-canary.3 again.  @marcklingen

**Comment by YYsuni** [comment]
For creating a object, you can view https://github.com/YYsuni/react18-json-view?tab=readme-ov-file#editable. And others, I will take care of that.

**Comment by YYsuni** [comment]
On @0.2.8-canary.4, you could delete the root node and reset it, @marcklingen

**Comment by YYsuni** [comment]
@marcklingen Hi?

**Comment by marcklingen** [comment]
Hi @YYsuni, I tried it again a couple of times but then got a lot of feedback by our users who edit a lot of JSON within our application. We switched to codemirror for editing and use react18-json-viewer for viewing of JSON across the UI. `react18-json-viewer` is awesome when editing single values but not when writing large JSON objects. Having multi-cursors and other editor features is preferred for that use case.
